### PR TITLE
Fix bug in JoinGraphBuilder

### PR DESCRIPTION
**Pull request by mrzzzrm**
Fix a bug in the `JoinGraphBuilder` where subgraphs not connected by any predicates would leave the `JoinGraph` unconnected. This PR makes the `JoinGraphBuilder` insert unpredicated edges between the components to create a connected `JoinGraph`

### CODE REVIEW

File: src/lib/optimizer/join_ordering/join_graph_builder.cpp [hunk 1]
```diff
@@ -169,4 +177,109 @@ JoinGraphBuilder::PredicateParseResult JoinGraphBuilder::_parse_union(
...
+  while (!remaining_vertex_indices.empty()) {
+    const auto vertex_idx = *remaining_vertex_indices.begin();
+
+    one_vertex_per_component.emplace_back(vertex_idx);
+
+    std::stack<size_t> bfs_stack;
+    bfs_stack.push(vertex_idx);
+
+    while (!bfs_stack.empty()) {
+      const auto vertex_idx2 = bfs_stack.top();
```
Comment (mrks) [inline src/lib/optimizer/join_ordering/join_graph_builder.cpp]: Can we find a better name for the two indexes?

File: src/lib/optimizer/join_ordering/join_graph_builder.cpp [hunk 2]
```diff
@@ -169,4 +177,109 @@ JoinGraphBuilder::PredicateParseResult JoinGraphBuilder::_parse_union(
...
+      bfs_stack.pop();
```
Comment (mrks) [inline src/lib/optimizer/join_ordering/join_graph_builder.cpp]: Breadth-First? I think I could live with the abbreviation, but a comment would be good.

File: src/lib/optimizer/join_ordering/join_graph_builder.cpp [hunk 3]
```diff
@@ -169,4 +177,109 @@ JoinGraphBuilder::PredicateParseResult JoinGraphBuilder::_parse_union(
...
+   *
+   * where the edges AD and DF are being created and have no predicates. There is of course the theoretical chance that
+   * different edges, say CD and EF would result in a better plan. We ignore this possibility for now.
+   */
+
+  std::unordered_set<size_t> remaining_vertex_indices;
+  for (auto vertex_idx = size_t{0}; vertex_idx < vertices.size(); ++vertex_idx)
+    remaining_vertex_indices.insert(vertex_idx);
+
+  std::vector<size_t> one_vertex_per_component;
```
Comment (mrks) [inline src/lib/optimizer/join_ordering/join_graph_builder.cpp]: What does this store?

**Comment by mrzzzrm** [comment]
Addressed your comments and then some. The long time it takes to get an review has the pro side that you're forced to look at older code and then understand it again. Added quite some comments and improved naming.

**Comment by mrks** [comment]
Haha, you don't want to provide me with an excuse for leaving these for a longer time ;)

**Pull request merged**